pulsar-client/src/main/java/com/yahoo/pulsar/client/api/Consumer.java

```diff
@@ -231,4 +233,12 @@
      * breaks, the messages are redelivered after reconnect.
      */
     void redeliverUnacknowledgedMessages();
+
+    /**
+     * Redelivers the given unacknowledged messages. In Failover mode, the request is ignored if the consumer is not
+     * active for the given topic. In Shared mode, the consumers messages to be redelivered are distributed across all
+     * the connected consumers. This is a non blocking call and doesn't throw an exception. In case the connection
+     * breaks, the messages are redelivered after reconnect.
+     */
+    void redeliverUnacknowledgedMessages(List<MessageIdImpl> messageIds);
```

Comment:
@sboobna does it really make sense to receive `MessageId`? It's just an empty interface